ui-modules/blueprint-composer/app/components/providers/blueprint-service.provider.js

```diff
@@ -603,16 +616,24 @@ function BlueprintService($log, $q, $sce, paletteApi, iconGenerator, dslService)
 
     function populateLocationFromApiSuccess(entity, data) {
         entity.clearIssues({group: 'location'});
-        entity.location = data.symbolicName;
-        entity.miscData.set('locationName', data.name);
-        entity.miscData.set('locationIcon', data.iconUrl || iconGenerator(data.symbolicName));
+        entity.location = data.yamlHere || data.symbolicName;
+        
+        let name = data.name || data.displayName;
+        if (!name && data.yamlHere) {
+            name = typeof data.yamlHere === 'object' ? Object.keys(data.yamlHere)[0] : data.yamlHere;
+        }
+        if (!name) name =  data.symbolicName;
+        entity.miscData.set('locationName', name);
+        
+        // use icon on item, but if none then generate using *yaml* to distinguish when someone has changed it
+        // (especially for things like jclouds:aws-ec2 -- the config is more interesting than the type name)
+        entity.miscData.set('locationIcon', data==null ? null : data.iconUrl || iconGenerator(data.yamlHere ? JSON.stringify(data.yamlHere) : data.symbolicName));
         return entity;
     }
 
     function populateLocationFromApiError(entity) {
-        entity.clearIssues({group: 'location'});
+        populateLocationFromApiSuccess(entity, { yamlHere: entity.location });
```

Comment:
That's wrong. If the location can be setup from the entity, then this code should be put into the caller function, not here. This function is to setup the state if there is an error when setting up the location.

Also, you want to keep the `entity.clearIssues({group: 'location'});` because errors are recreated just below. Not doing so will result to the same error displayed twice

Reply:
from a UX point of view, if there is an error we should indicate this fact (which this method now does) but it should still make best-effort to reflect what is known about the location in the UI (which the call to `populateOnSuccess` does), in addition to that error.  the call to `populateOnSuccess` clears the issues (so we don't get double), it also sets the ` `locationName` sensibly, so by calling that now (the changes I make here) we avoid code duplication.  i think the behaviour as changed is best.

Reply:
Hum, I get your point. I think I'm complaining mainly because the name is not right, this should be `populateLocation` instead of `populateLocationFromApiSuccess`. Could you change that?

Reply:
:) yes, i concur!